lib/dns.js

```diff
@@ -189,6 +189,9 @@ exports.lookupService = function(host, port, callback) {
   if (cares.isIP(host) === 0)
     throw new TypeError('"host" argument needs to be a valid IP address');
 
+  if (typeof port !== 'number')
```

Comment:
I definitely think coercion of numbers should be discussed first. I also think that this PR should go forward since it fixes a bug regardless of that issue (non-number ports aren't accepted as of today).